packages/teleport/src/Discover/SelectResource/SelectResource.test.tsx

```diff
@@ -0,0 +1,337 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+
+import { MemoryRouter } from 'react-router';
+
+import { render } from 'design/utils/testing';
+
+import { SelectResource } from 'teleport/Discover/SelectResource/SelectResource';
+import { Access, Acl, makeUserContext } from 'teleport/services/user';
+
+const fullAccess: Access = {
+  list: true,
+  read: true,
+  edit: true,
+  create: true,
+  remove: true,
+};
+
+const fullAcl: Acl = {
+  windowsLogins: ['Administrator'],
+  tokens: fullAccess,
+  appServers: fullAccess,
+  kubeServers: fullAccess,
+  recordedSessions: fullAccess,
+  activeSessions: fullAccess,
+  authConnectors: fullAccess,
+  roles: fullAccess,
+  users: fullAccess,
+  trustedClusters: fullAccess,
+  events: fullAccess,
+  accessRequests: fullAccess,
+  billing: fullAccess,
+  dbServers: fullAccess,
+  desktops: fullAccess,
+  nodes: fullAccess,
+  clipboardSharingEnabled: true,
+  desktopSessionRecordingEnabled: true,
+  directorySharingEnabled: true,
+  connectionDiagnostic: fullAccess,
+};
+
+const userContextJson = {
+  authType: 'sso',
+  userName: 'Sam',
+  accessCapabilities: {
+    suggestedReviewers: ['george_washington@gmail.com', 'chad'],
+    requestableRoles: ['dev-a', 'dev-b', 'dev-c', 'dev-d'],
+  },
+  cluster: {
+    name: 'aws',
+    lastConnected: '2020-09-26T17:30:23.512876876Z',
+    status: 'online',
+    nodeCount: 1,
+    publicURL: 'localhost',
+    authVersion: '4.4.0-dev',
+    proxyVersion: '4.4.0-dev',
+  },
+};
+
+describe('select resource', () => {
+  function create(resource: string, userAcl: Acl) {
+    const userContext = makeUserContext({
+      ...userContextJson,
+      userAcl,
+    });
+
+    return render(
+      <MemoryRouter initialEntries={[{ state: { entity: resource } }]}>
+        <SelectResource
+          userContext={userContext}
+          isEnterprise={false}
+          nextStep={() => null}
+        />
+      </MemoryRouter>
+    );
+  }
+
+  describe('server', () => {
+    test('shows permissions error when lacking tokens.create', () => {
+      const result = create('server', {
+        ...fullAcl,
+        tokens: {
+          ...fullAccess,
+          create: false,
+        },
+      });
+
+      const permissionsError = result.getByTestId('permissions-error');
+      expect(permissionsError).toBeInTheDocument();
+
+      expect(result.container.firstChild).toMatchSnapshot();
+
+      const proceedButton = result.getByTestId('proceed-button');
+
+      expect(proceedButton.hasAttribute('disabled')).toBe(true);
+    });
+
+    test('shows permissions error when lacking nodes.read', () => {
```

Comment:
```suggestion
    test('shows permissions error when lacking nodes.list', () => {
```

node is an odd one in that it only requires `list`

Reply:
Cheers, updated